### PR TITLE
Issue #1163 - Convert Dessert to use templates

### PR DESCRIPTION
**Pull request by Mailkov**
IA Page: https://duck.co/ia/view/dessert

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@Mailkov this is now deployed to beta!

**Comment by GuiltyDolphin** [comment]
@Mailkov I'm practically drooling looking at the code for this!

File: lib/DDG/Goodie/Dessert.pm [hunk 1]
```diff
@@ -85,14 +92,14 @@ sub begins_with {
...
     # Check if a value exists given our key.
     if(exists $desserts{$letter}) {
-	my $value = $desserts{$letter};
+	    my $value = $desserts{$letter};
 
-	# Check if a certain dessert begins with our query string.
-	for my $dessert (@{$value}) {
-	    if($dessert =~ /^$query/i) {
-		push @items, $dessert;
+	    # Check if a certain dessert begins with our query string.
```
Comment (mintsoft) [inline lib/DDG/Goodie/Dessert.pm]: @Mailkov nit: there's a mixture of hard-tabs and spaces here

File: lib/DDG/Goodie/Dessert.pm [hunk 2]
```diff
@@ -29,38 +20,53 @@ sub itemify {
...
+    structured_answer => {
+        id => 'dessert',
+        name => 'Answer',
+        data => {
+            title => $dessert,
+            subtitle => $end,
+            url => "http://duckduckgo.com/?q=$dessert_link+recipe"
+        },
+        templates => {
+            group => 'info',
```
Comment (GuiltyDolphin) [inline lib/DDG/Goodie/Dessert.pm]: How about using the `text` group?

**Comment by moollaza** [comment]
Thanks! This LGTM :+1:

**Pull request merged**